tests/static_map/insert_or_apply_test.cu

```diff
@@ -33,6 +33,14 @@
 
 using size_type = std::size_t;
 
+struct binary_plus_op {
+  template <typename T, cuda::thread_scope Scope>
+  __device__ void operator()(cuda::atomic_ref<T, Scope> lhs, T rhs)
+  {
+    lhs.fetch_add(rhs, cuda::memory_order_relaxed);
+  }
+};
+
```

Comment:
I think we should provide a set of operators so users don't have to implement them all the time. For instance, all of the supported atomic operations.

Could be a separate PR.

Reply:
Agreed! I can make a seperate PR to add reduction operators for most basic commonly used reductions.